### DISCUSSION TITLE
user_config_override.h // Tasmocompiler

### DISCUSSION THREAD

**Discussion by poseidogithub** [arendst/Tasmota#17510] [Q&A] [answered]
Hello!

I'm trying to compile tasmota to add OpenTherm suport using Tasmocompiler.
I was able to compile it, flash it on my Wemos D1 and it's working with my boiler.
Now I want to go a step further and modify the defaults and limits on the temperatures and following this info(https://github.com/arendst/Tasmota/issues/16880#issuecomment-1287882580)  I've added this to the parameters in tasmocompiler:

#ifndef USE_OPENTHERM
#define USE_OPENTHERM
#endif

#define OT_HOT_WATER_MIN 23
#define OT_HOT_WATER_MAX 40
#define OT_BOILER_MIN 25
#define OT_BOILER_MAX 35

but the values defined seem not to be taken in to consideration.... I'm not a programmer but I suspect that I should "open" and "close" something before and anfter the define?

thanks in advance

**Comment by poseidogithub** [comment, marked as answer]
for others reference: seems that the code should be:


#ifndef USE_OPENTHERM
#define USE_OPENTHERM
#endif



#ifndef OT_HOT_WATER_MIN
#define OT_HOT_WATER_MIN 23
#endif
#ifndef OT_HOT_WATER_MAX
#define OT_HOT_WATER_MAX 39
#endif
#ifndef OT_BOILER_MIN
#define OT_BOILER_MIN 25
#endif
#ifndef OT_BOILER_MAX
#define OT_BOILER_MAX 35
#endif

#ifndef OT_HOT_WATER_DEFAULT
#define OT_HOT_WATER_DEFAULT 36;
#endif
#ifndef OT_BOILER_DEFAULT
#define OT_BOILER_DEFAULT 30;
#endif

**Reply by barbudor** [reply]
Your original code was correct
The new one is bad, especially the wrong `;` in the last two (Yes there are some in the original code, but they shouldn't be there. It's working just by luck)

The proper way should be (although as I wrote above, your initial version was still good):
```
#ifdef OT_HOT_WATER_MIN
#undef OT_HOT_WATER_MIN
#endif
#define OT_HOT_WATER_MIN 23
```
Note that I'm using #ifdef so that if the symbol is already defined, it is then undefined before redefining with your custom value


Why do you say they are not taken into account ?
Is that normal that the boiler range is within the water range ?
In the original code, the boiler range is much above the water range

**Reply by sfromis** [reply]
When you make build time changes of *defaults* for settings also available to be modified via commands, those defaults will only have effect after a complete reset of Tasmota to initial defaults as of your build. After first boot, the persistent Tasmota settings structure containing current values (even if unchanged by command) will be used each boot, meaning that build time changes makes no difference to settings without the reset.
